### PR TITLE
Material: Continued Material enahncements

### PR DESCRIPTION
**Pull request by davesrocketshop**
Continues the work of the material subsystem improvements.

Several important items are included in this merge. In terms of new capabilities, this merge adds List and MultiLineString as valid property types, complete with editing dialogs. This will help with backwards compatibility for external workbenches, such as Render.

Stability has been a big focus. New unit tests help to verify features work as expected. Bugs have been fixed and crashes avoided.

Material cards have had a renaming to their tree structure. For example, 'StandardMeterials' is redundant, so this was renamed to 'Standard'. The cards themselves are more compliant fully passing the yamllint tests.

More soon.

### CODE REVIEW

**Comment by davesrocketshop** [comment]
I honestly don't understand how git keeps changing files in other modules. I'm not changing code there.

**Comment by chennes** [comment]
How are you doing the rebase to upstream?

**Comment by davesrocketshop** [comment]
git pull --rebase upstream main

In this case it made me review over 80 merges manually. I'm basically abandoning that dev branch and will be starting a new one once this is merged.

File: src/Mod/Material/App/MaterialLibrary.cpp [hunk 1]
```diff
@@ -42,8 +42,8 @@ using namespace Materials;
 
 TYPESYSTEM_SOURCE(Materials::MaterialLibrary, LibraryBase)
 
-MaterialLibrary::MaterialLibrary()
-{}
+// MaterialLibrary::MaterialLibrary()
+// {}
```
Comment (quarnster) [inline src/Mod/Material/App/MaterialLibrary.cpp:46]: Any reason to keep this?

File: src/Mod/Material/App/MaterialLibrary.cpp [hunk 2]
```diff
@@ -364,8 +365,8 @@ MaterialLibrary::getMaterialTree() const
 
 TYPESYSTEM_SOURCE(Materials::MaterialExternalLibrary, MaterialLibrary::MaterialLibrary)
 
-MaterialExternalLibrary::MaterialExternalLibrary()
-{}
+// MaterialExternalLibrary::MaterialExternalLibrary()
+// {}
```
Comment (quarnster) [inline src/Mod/Material/App/MaterialLibrary.cpp:369]: And this?

**Comment by davesrocketshop** [comment]
Nope. But given the git issues I've been having I'd prefer to defer minor edits like this to a new branch

**Comment by chennes** [comment]
It looks like the changes to TaskSketcherTool are whitespace-only -- can you remove them from this PR? Also, I'd check your git line ending settings (the `core.autocrlf` setting).

**Comment by davesrocketshop** [comment]
On my linux it's unset. On Windows it's:

file:C:/Program Files/Git/etc/gitconfig core.autocrlf=true

I'll revert the file shortly

**Comment by davesrocketshop** [comment]
Reversion, and minor edits, complete

File: src/Mod/Material/App/MaterialLibrary.cpp (hunk 1 above)
Comment (davesrocketshop) [reply src/Mod/Material/App/MaterialLibrary.cpp]: Removed

File: src/Mod/Material/App/MaterialLibrary.cpp (hunk 2 above)
Comment (davesrocketshop) [reply src/Mod/Material/App/MaterialLibrary.cpp]: Removed

**Pull request merged**